### DISCUSSION TITLE
Getting 404 Error on Questionset Comment Update Endpoint

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-inQuiry/Community#115] [General Discussions]
We have set up the inQuiry and Knowlg services on AWS, and most of the APIs seem to be working. However, we're encountering a 404 Not Found error for the following endpoint in the inQuiry service:

PATCH `/questionset/v5/comment/update/do_id`
Note - Same do_id works for other questionset APIs, so it looks like error that route not found.

The services were installed using the Ed-installer from the release-7.0.0 branch:
[Ed-installer release-7.0.0](https://github.com/project-sunbird/sunbird-ed-installer/tree/release-7.0.0)
As per the [Ed release notes](https://ed.sunbird.org/use/release/release-notes#:~:text=Sunbird%2DinQuiry%3A%20V7.0.0), inQuiry v7.0.0 is being used.

I have also verified that this endpoint exists in the assessment service route:
[Assessment route reference](https://github.com/Sunbird-inQuiry/inquiry-api-service/blob/release-7.0.0/assessment-api/assessment-service/conf/routes#L78)

Could you help us understand why we're receiving a 404 error for the /questionset/comment/update/do_id endpoint?

@kiranharidas187 @ramkumar-pacewisdom  @Vivek-M-08
I want to understand that in which release of inQuiry the comment update is added in inQuiry service.
Editor side I remember it was implemented in release-8.0.0, but the backend side was it not done in release-7.0.0 ?

@pallakartheekreddy @krgauraw 
@AbhilashKD @coolbung @snehal0904 @itsvick

**Comment by Vivek-M-08** [comment]
@rajnishdargan - Comment update is added in release-7.0.0 for inQuiry service 
here is the PR for reference: https://github.com/Sunbird-inQuiry/inquiry-api-service/pull/49

**Reply by rajnishdargan** [reply]
@Vivek-M-08 
Could you please confirm if this is the correct payload for making the API call?

PATCH `/questionset/v5/comment/update/do_id`

```
{
    "request": {
        "comments": [
            {
                "comment": "Test"
            }
        ]
    }

```
This API is not present in inQuiry microsite API documentation.

**Reply by Vivek-M-08** [reply]
Can you please try the following request body once 
```
{
  "request":{
      "comments" : [
          {
            "identifier" : "do_11388216691058278411",
            "comment" : "Comments made by the reviewer"
          }
      ]
  }
}
```

**Reply by rajnishdargan** [reply]
Thanks @Vivek-M-08 
This kind of payload is working..
Attaching the sample CURL for reference:

```
curl --location --request PATCH '{{assessment-service-url}}/questionset/v5/comment/update' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "comments": [
            {
                "identifier": "do_id",
                "comment": "Comments made by the reviewer"
            }
        ]
    }
}'
```
